startup/00-startup.py

```diff
@@ -1,9 +1,7 @@
 from nslsii import configure_base
 from IPython import get_ipython
 
-configure_base(get_ipython().user_ns, 'local',
-               configure_logging=False,
-               ipython_exc_logging=False)
+configure_base(get_ipython().user_ns, 'local')
```

Comment:
I couldn't get this to run using an older version of `nslsii` (version 0.0.14) unless I expanded `configure_base()`. Should this be in a try except block? Something like:
```
try:
    configure_base(get_ipython().user_ns, 'local')
except FileNotFoundError:
    configure_base(get_ipython().user_ns, 'local',
                   configure_logging=False,
                   ipython_exec_logging=False)
```

Reply:
I think it should work for both versions :). Thanks for looking into it. I'll clean up the code I intended to remove before merging and will add the suggested update.

Reply:
What file is it complaining about? Could you please post the traceback?

Reply:
Yeah, here it is.
```
(testopt37) C:\Users\Abby>ipython --profile=simulated_hardware
Python 3.7.0 (default, Jun 28 2018, 08:04:48) [MSC v.1912 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.18.1 -- An enhanced Interactive Python. Type '?' for help.

IPython profile: simulated_hardware
C:\Users\Abby\Anaconda3\envs\testopt37\lib\site-packages\pims\image_reader.py:26: RuntimeWarning: PIMS image_reader.py could not find scikit-image. Falling back to matplotlib's imread(), which uses floats instead of integers. This may break your scripts.
(To ignore this warning, include the line "warnings.simplefilter("ignore", RuntimeWarning)" in your script.)
  warnings.warn(RuntimeWarning(ski_preferred))
environment variable BLUESKY_LOG_FILE is not set, using default file path
bluesky log file: /var/log/bluesky/bluesky.log
[TerminalIPythonApp] WARNING | Unknown error in handling startup files:
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
~\Anaconda3\envs\testopt37\lib\site-packages\IPython\core\shellapp.py in _exec_file(self, fname, shell_futures)
    350                                                  self.shell.user_ns,
    351                                                  shell_futures=shell_futures,
--> 352                                                  raise_exceptions=True)
    353         finally:
    354             sys.argv = save_argv

~\Anaconda3\envs\testopt37\lib\site-packages\IPython\core\interactiveshell.py in safe_execfile(self, fname, exit_ignore, raise_exceptions, shell_futures, *where)
   2739                 py3compat.execfile(
   2740                     fname, glob, loc,
-> 2741                     self.compile if shell_futures else None)
   2742             except SystemExit as status:
   2743                 # If the call was made with 0 or None exit status (sys.exit(0)

~\Anaconda3\envs\testopt37\lib\site-packages\IPython\utils\py3compat.py in execfile(fname, glob, loc, compiler)
    166     with open(fname, 'rb') as f:
    167         compiler = compiler or compile
--> 168         exec(compiler(f.read(), fname, 'exec'), glob, loc)
    169
    170 # Refactor print statements in doctests.

~\.ipython\profile_simulated_hardware\startup\00-startup.py in <module>
      9 #                   ipython_exc_logging=False)
     10
---> 11 configure_base(get_ipython().user_ns, 'local')
     12
     13 bec.disable_plots()

~\Anaconda3\envs\testopt37\lib\site-packages\nslsii\__init__.py in configure_base(user_ns, broker_name, bec, epics_context, magics, mpl, configure_logging, pbar, ipython_exc_logging)
    188
    189     if configure_logging:
--> 190         configure_bluesky_logging(ipython=get_ipython())
    191
    192     if ipython_exc_logging:

~\Anaconda3\envs\testopt37\lib\site-packages\nslsii\__init__.py in configure_bluesky_logging(ipython)
    280     print(f"bluesky log file: {bluesky_log_file_path}", file=sys.stderr)
    281     log_file_handler = TimedRotatingFileHandler(
--> 282         filename=bluesky_log_file_path, when="W0", backupCount=10
    283     )
    284     log_file_handler.setLevel("INFO")

~\Anaconda3\envs\testopt37\lib\logging\handlers.py in __init__(self, filename, when, interval, backupCount, encoding, delay, utc, atTime)
    197     """
    198     def __init__(self, filename, when='h', interval=1, backupCount=0, encoding=None, delay=False, utc=False, atTime=None):
--> 199         BaseRotatingHandler.__init__(self, filename, 'a', encoding, delay)
    200         self.when = when.upper()
    201         self.backupCount = backupCount

~\Anaconda3\envs\testopt37\lib\logging\handlers.py in __init__(self, filename, mode, encoding, delay)
     52         Use the specified filename for streamed logging
     53         """
---> 54         logging.FileHandler.__init__(self, filename, mode, encoding, delay)
     55         self.mode = mode
     56         self.encoding = encoding

~\Anaconda3\envs\testopt37\lib\logging\__init__.py in __init__(self, filename, mode, encoding, delay)
   1039             self.stream = None
   1040         else:
-> 1041             StreamHandler.__init__(self, self._open())
   1042
   1043     def close(self):

~\Anaconda3\envs\testopt37\lib\logging\__init__.py in _open(self)
   1068         Return the resulting stream.
   1069         """
-> 1070         return open(self.baseFilename, self.mode, encoding=self.encoding)
   1071
   1072     def emit(self, record):

FileNotFoundError: [Errno 2] No such file or directory: 'C:\\var\\log\\bluesky\\bluesky.log'

Reply:
Thanks! What happens if the directory `C:\\var\\log\\bluesky\\` is pre-created in advance?

Reply:
The error is gone after creating the directory. Looks like later versions of `nslsii` create the directory if it doesn't exist, but the version I was using doesn't.

Reply:
I see. Yes, the latest version creates the log directory: https://github.com/NSLS-II/nslsii/blob/79ea5b85121619ce6bc65dc961fb33a8e604460f/nslsii/__init__.py#L313. I think we can leave try/except for now.